src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

```diff
@@ -0,0 +1,192 @@
+    /// implementations the parameters will be JSON serialized using System.Text.Json and passed as a parameter to the
+    /// underlying JavaScript implementation that handles the operation details.
+    /// </remarks>
+    public bool TryAddAdditionalParameter<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string name, TValue value)
+    {
+        ArgumentNullException.ThrowIfNull(name, nameof(name));
+        AdditionalRequestParameters ??= new();
+        return AdditionalRequestParameters.TryAdd(name, value);
+    }
+
+    /// <summary>
+    /// Tries to remove an existing additional parameter.
+    /// </summary>
+    public bool TryRemoveAdditionalParameter<TValue>(string name)
+    {
+        ArgumentNullException.ThrowIfNull(name, nameof(name));
+        return AdditionalRequestParameters != null && AdditionalRequestParameters.Remove(name);
+    }
+
+    /// <summary>
+    /// Tries to retrieve an existing additional parameter.
+    /// </summary>
+    public bool TryGetAdditionalParameter<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string name, out TValue value)
+    {
+        ArgumentNullException.ThrowIfNull(name);
+
+        value = default;
+        if (AdditionalRequestParameters == null || !AdditionalRequestParameters.TryGetValue(name, out var rawValue))
+        {
+            return false;
+        }
+        if (rawValue is JsonElement json)
+        {
+            value = Deserialize(json);
+            AdditionalRequestParameters[name] = value;
+            return true;
+        }
+        else
+        {
+            value = (TValue)rawValue;
+            return true;
+        }
+
+        [UnconditionalSuppressMessageAttribute(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "All the types serialized and deserialized have annotated APIs on the parent method that ensure the members are preserved.")]
+        static TValue Deserialize(JsonElement element) => element.Deserialize<TValue>();
+    }
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Serializes 'InteractiveAuthenticationRequest' into a string")]
+    internal string ToState() => JsonSerializer.Serialize(this);
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Deserializes InteractiveAuthenticationRequest")]
+    internal static InteractiveRequestOptions FromState(string state) => JsonSerializer.Deserialize<InteractiveRequestOptions>(state);
```

Comment:
You might be able to avoid all these problems with linker warnings and suppressions if you use the source-generated version of JSON serialization instead. That's what it's there for.

For example, declare a type like 

```cs
// Our custom JSON serializer context that generates code to serialize
// arrays of planets so that we can use it with our HTTP endpoint.
[JsonSerializable(typeof(InteractiveRequestOptions))]
public partial class InteractiveRequestOptionsSerializerContext : JsonSerializerContext
{
}
```

... and then when calling JsonSerializer.Deserialize or JsonSerializer.Serialize, pass an instance of InteractiveRequestOptionsSerializerContext as the "context" argument.

Reply:
More examples: https://blog.martincostello.com/using-json-source-generators-with-aspnet-core-minimal-apis/

Reply:
If this can be done internally, then we can definitely do that, let me see if I can make it work.

Reply:
I switched to use the serialization context